### ISSUE TITLE
Eliminate `FluentLogger` from our API

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#1526]
The following classes generated build time ErrorProne warning of [`NonApiType`](https://errorprone.info/bugpattern/NonApiType):
 * `ErrorHandler`
 * `ConsumerErrorHander`
 * `Subscriptions`
 * `LoggingHandlerWithType`
 * LoggingHandler`
 * `LoggingServerErrorHandler`
 * `LoggingTypeErrorHandler`
 * `LoggingErrorHandler`
 * `LoggingConsumerErrorHandler`
 * `ReceptorLog`
 * `LoggingHandlerWithType`

The message is
```
 warning: [NonApiType] There is no advantage to passing around a logger rather than declaring one in the class that needs it. 
    LoggingConsumerErrorHandler(FluentLogger logger,
                                             ^
    (see https://errorprone.info/bugpattern/NonApiType)
```

There are two issues associated with this:
  1. We do not want to use `FluentLogger` in our API because we have our Logging library.
  2. We need to address the nature of this ErrorProne warning in our API.

**Comment by alexander-yevsyukov** [comment]
@yevhenii-nadtochii, FYI.

**Comment by alexander-yevsyukov** [comment]
Addressed by #1529.
